### PR TITLE
[deps] Update dependencies to latest versions / fix examples

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
1. I've fixed all the examples, and now they all run including the javascript ones.  We will likely want to at least have one ESM example again, just to show how a proper setup would look like for it.
2. All dependencies are updated to the latest including Typescript!  We may want to move some to peerDependencies and allow people to choose their own versions to not deal with conflicts.
3. `param?: undefined` is gone!  The newer versions of typescript are much more strict / didn't like this, so I've properly changed it to a type based overload.  It's additionally been tested in Javascript to work!
4. GraphQL had to be regenerated with the updated generation dependency

### Test Plan
pnpm test, for both examples and the unit tests, CI should also check them as well

### CODE REVIEW

**Comment by 0xjinn** [comment]
Looks like there is alot of timeouts with local indexer?

**Comment by gregnazario** [comment]
> Looks like there is alot of timeouts with local indexer?

That's been happening to me a bunch in CI @banool

File: src/client/get.ts [hunk 1]
```diff
@@ -47,7 +47,9 @@ export type GetAptosRequestOptions = Omit<GetRequestOptions, "type">;
  * @param options GetRequestOptions
  * @returns
  */
-export async function get<Req, Res>(options: GetRequestOptions): Promise<AptosResponse<Req, Res>> {
+export async function get<Req extends {}, Res extends {}>(
```
Comment (0xmaayan) [inline src/client/get.ts:50]: why do you add extends everywhere..?

Comment (gregnazario) [reply src/client/get.ts:50]: The generics have to be objects, rather than just any input. There's stronger typing in future versions of some of these

**Review by 0xmaayan — COMMENTED** [review]
lots of files. can we test it locally to make sure it works with nextjs/node/web app? can link a local build.

**Comment by gregnazario** [comment]
> lots of files. can we test it locally to make sure it works with nextjs/node/web app? can link a local build.

For node, the examples are now automated in CI.

NextJS / webapp, I'll have to build one

**Comment by 0xmaayan** [comment]
> > lots of files. can we test it locally to make sure it works with nextjs/node/web app? can link a local build.
> 
> For node, the examples are now automated in CI.
> 
> NextJS / webapp, I'll have to build one

Great. I am also curios what would happen for project that use different/lower ts versions.

**Review by 0xmaayan — APPROVED** [review]
I tested it doesnt break nextjs/node/web app with a very basic indexer/fullnode queries. i.e just run the app and make those queries.

Also tested with a local wallet adapter demo app with sdk v2 support.

Let's roll with it but keep an eye if issues are coming

**Pull request merged**